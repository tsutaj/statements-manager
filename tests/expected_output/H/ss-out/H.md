


## H: Many Kinds of Apples

### Problem Statement

Apple Farmer Mon has two kinds of tasks: "harvest apples" and "ship apples".

There are $N$ different species of apples, and $N$ distinguishable boxes. Apples are labeled by the species, and boxes are also labeled, from $1$ to $N$. The $i$-th species of apples are stored in the $i$-th box.

For each $i$, the $i$-th box can store at most $c_i$ apples, and it is initially empty (no apple exists).

Mon receives $Q$ instructions from his boss Kukui, and Mon completely follows in order. Each instruction is either of two types below.

- "harvest apples": put $d$ $x$-th apples into the $x$-th box.
- "ship apples": take $d$ $x$-th apples out from the $x$-th box.

However, not all instructions are possible to carry out. Now we call an instruction which meets either of following conditions "impossible instruction":

- When Mon harvest apples, the amount of apples exceeds the capacity of that box.
- When Mon tries to ship apples, there are not enough apples to ship.

Your task is to detect the instruction which is impossible to carry out.

### Input

Input is given in the following format.

```
$N$
$c_1$ $c_2$ $\cdots$ $c_N$
$Q$
$t_1$ $x_1$ $d_1$
$t_2$ $x_2$ $d_2$
$\vdots$
$t_Q$ $x_Q$ $d_Q$
```

In line 1, you are given the integer $N$, which indicates the number of species of apples.

In line 2, given $c_i$ $(1 \leq i \leq N)$ separated by whitespaces. $c_i$ indicates the capacity of the $i$-th box.

In line 3, given $Q$, which indicates the number of instructions.
Instructions are given successive $Q$ lines. $t_i$ $x_i$ $d_i$ means what kind of instruction, which apple Mon handles in this instruction, how many apples Mon handles, respectively. If $t_i$ is equal to $1$, it means Mon does the task of "harvest apples", else if $t_i$ is equal to $2$, it means Mon does the task of "ship apples".

### Constraints

All input values are integers, and satisfy the following constraints.

- $1 \leq N \leq 1{,}000$
- $1 \leq c_i \leq 100{,}000$ $(1 \leq i \leq N)$
- $1 \leq Q \leq 100{,}000$
- $t_i \in \left\\{ 1, 2 \right\\}$ $(1 \leq i \leq Q)$
- $1 \leq x_i \leq N$ $(1 \leq i \leq Q)$
- $1 \leq d_i \leq 100{,}000$ $(1 \leq i \leq Q)$

### Output

If there is "impossible instruction", output the index of the apples which have something to do with the first "impossible instruction".
Otherwise, output $0$.


### Sample Input 1
```
2
3 3
4
1 1 2
1 2 3
2 1 3
2 2 3
```



### Sample Output 1
```
1
```





In this case, there are not enough apples to ship in the first box.


### Sample Input 2
```
2
3 3
4
1 1 3
2 1 2
1 2 3
1 1 3
```



### Sample Output 2
```
1
```





In this case, the amount of apples exceeds the capacity of the first box.


### Sample Input 3
```
3
3 4 5
4
1 1 3
1 2 3
1 3 5
2 2 2
```



### Sample Output 3
```
0
```





### Sample Input 4
```
6
28 56 99 3 125 37
10
1 1 10
1 1 14
1 3 90
1 5 10
2 3 38
2 1 5
1 3 92
1 6 18
2 5 9
2 1 4
```



### Sample Output 4
```
3
```










