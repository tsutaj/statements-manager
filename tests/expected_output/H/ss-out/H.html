<!DOCTYPE html>
<html lang="ja">

<head>
    <title>Problem Statement</title>
    <meta charset="utf-8">
    <link href='http://fonts.googleapis.com/css?family=Noto+Serif' rel='stylesheet' type='text/css'>
    <style type="text/css">
        html {
            color: #222222;
            font-family: "Noto Serif";
        }

        h1 {
            font-size: 16pt;
            border-bottom: solid 2px #e0e0e0;
            margin-bottom: 20px;
        }

        h2 {
            font-size: 14pt;
            border-bottom: solid 2px #e0e0e0;
            margin-bottom: 20px;
            margin-top: 20px;
        }

        h3 {
            font-size: 14pt;
            border-bottom: solid 2px #e0e0e0;
            margin-bottom: 20px;
            margin-top: 20px;
        }

        h4 {
            font-size: 13pt;
            margin-bottom: 20px;
            margin-top: 20px;
        }

        h5 {
            font-size: 12pt;
            margin-bottom: 20px;
            margin-top: 20px;
        }

        b {
            font-weight: bold;
        }

        p {
            font-size: 12pt;
        }

        li {
            font-size: 12pt;
        }

        pre {
            font-size: 11pt;
            padding: 8px;
            border: solid 1px #e0e0e0;
            border-radius: 4px;
        }

        code {
            color: #e83e8c;
        }

        code.language-input-format {
            color: #222222;
        }

        .testcase_pre {
            font-size: 11pt;
            padding: 8px;
            border: solid 1px #e0e0e0;
            border-radius: 4px;
            overflow: auto;
            max-width: 480;
            max-height: 340;
        }

        .error_pre {
            font-size: 11pt;
            padding: 8px;
            border: solid 1px #f0e0e0;
            border-radius: 4px;
            overflow: auto;
            max-height: 340;
            background-color: #ffeeee;
        }

        .table tbody>tr>td.vert-aligned {
            vertical-align: middle;
        }
    </style>
    <script type="text/x-mathjax-config">
        MathJax.Hub.Config({
            tex2jax: {
                inlineMath: [["$", "$"]],
                processEscapes: true,
                processClass: 'input-format|language-input-format'
            }
        });
    </script>
    <script type="text/javascript"
        src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/MathJax.js?config=TeX-AMS_CHTML">
        </script>
</head>

<body>
    
    <div >
        <h2>H: Many Kinds of Apples</h2>
<h3>Problem Statement</h3>
<p>Apple Farmer Mon has two kinds of tasks: "harvest apples" and "ship apples".</p>
<p>There are $N$ different species of apples, and $N$ distinguishable boxes. Apples are labeled by the species, and boxes are also labeled, from $1$ to $N$. The $i$-th species of apples are stored in the $i$-th box.</p>
<p>For each $i$, the $i$-th box can store at most $c_i$ apples, and it is initially empty (no apple exists).</p>
<p>Mon receives $Q$ instructions from his boss Kukui, and Mon completely follows in order. Each instruction is either of two types below.</p>
<ul>
<li>"harvest apples": put $d$ $x$-th apples into the $x$-th box.</li>
<li>"ship apples": take $d$ $x$-th apples out from the $x$-th box.</li>
</ul>
<p>However, not all instructions are possible to carry out. Now we call an instruction which meets either of following conditions "impossible instruction":</p>
<ul>
<li>When Mon harvest apples, the amount of apples exceeds the capacity of that box.</li>
<li>When Mon tries to ship apples, there are not enough apples to ship.</li>
</ul>
<p>Your task is to detect the instruction which is impossible to carry out.</p>
<h3>Input</h3>
<p>Input is given in the following format.</p>
<pre class="input-format"><code class="language-input-format">$N$
$c_1$ $c_2$ $\cdots$ $c_N$
$Q$
$t_1$ $x_1$ $d_1$
$t_2$ $x_2$ $d_2$
$\vdots$
$t_Q$ $x_Q$ $d_Q$
</code></pre>
<p>In line 1, you are given the integer $N$, which indicates the number of species of apples.</p>
<p>In line 2, given $c_i$ $(1 \leq i \leq N)$ separated by whitespaces. $c_i$ indicates the capacity of the $i$-th box.</p>
<p>In line 3, given $Q$, which indicates the number of instructions.
Instructions are given successive $Q$ lines. $t_i$ $x_i$ $d_i$ means what kind of instruction, which apple Mon handles in this instruction, how many apples Mon handles, respectively. If $t_i$ is equal to $1$, it means Mon does the task of "harvest apples", else if $t_i$ is equal to $2$, it means Mon does the task of "ship apples".</p>
<h3>Constraints</h3>
<p>All input values are integers, and satisfy the following constraints.</p>
<ul>
<li>$1 \leq N \leq 1{,}000$</li>
<li>$1 \leq c_i \leq 100{,}000$ $(1 \leq i \leq N)$</li>
<li>$1 \leq Q \leq 100{,}000$</li>
<li>$t_i \in \left\{ 1, 2 \right\}$ $(1 \leq i \leq Q)$</li>
<li>$1 \leq x_i \leq N$ $(1 \leq i \leq Q)$</li>
<li>$1 \leq d_i \leq 100{,}000$ $(1 \leq i \leq Q)$</li>
</ul>
<h3>Output</h3>
<p>If there is "impossible instruction", output the index of the apples which have something to do with the first "impossible instruction".
Otherwise, output $0$.</p>
<h3>Sample Input 1</h3>
<pre class="input-format"><code class="language-input-format">2
3 3
4
1 1 2
1 2 3
2 1 3
2 2 3
</code></pre>
<h3>Sample Output 1</h3>
<pre class="input-format"><code class="language-input-format">1
</code></pre>
<p>In this case, there are not enough apples to ship in the first box.</p>
<h3>Sample Input 2</h3>
<pre class="input-format"><code class="language-input-format">2
3 3
4
1 1 3
2 1 2
1 2 3
1 1 3
</code></pre>
<h3>Sample Output 2</h3>
<pre class="input-format"><code class="language-input-format">1
</code></pre>
<p>In this case, the amount of apples exceeds the capacity of the first box.</p>
<h3>Sample Input 3</h3>
<pre class="input-format"><code class="language-input-format">3
3 4 5
4
1 1 3
1 2 3
1 3 5
2 2 2
</code></pre>
<h3>Sample Output 3</h3>
<pre class="input-format"><code class="language-input-format">0
</code></pre>
<h3>Sample Input 4</h3>
<pre class="input-format"><code class="language-input-format">6
28 56 99 3 125 37
10
1 1 10
1 1 14
1 3 90
1 5 10
2 3 38
2 1 5
1 3 92
1 6 18
2 5 9
2 1 4
</code></pre>
<h3>Sample Output 4</h3>
<pre class="input-format"><code class="language-input-format">3
</code></pre>
    </div>
    
</body>

</html>